src/ServiceControl/SagaAudit/SagaUpdatedHandler.cs

```diff
@@ -1,24 +1,102 @@
 ﻿namespace ServiceControl.SagaAudit
 {
+    using System;
+    using System.Collections.Concurrent;
+    using System.Threading;
     using System.Threading.Tasks;
     using EndpointPlugin.Messages.SagaState;
+    using Newtonsoft.Json.Linq;
     using NServiceBus;
-    using ServiceControl.Persistence;
+    using ServiceControl.Connection;
+    using ServiceControl.Infrastructure.DomainEvents;
 
     class SagaUpdatedHandler : IHandleMessages<SagaUpdatedMessage>
     {
-        public SagaUpdatedHandler(ISagaAuditDataStore store)
+        public SagaUpdatedHandler(IDomainEvents domainEvents, IPlatformConnectionBuilder connectionBuilder)
         {
-            this.store = store;
+            this.domainEvents = domainEvents;
+            this.connectionBuilder = connectionBuilder;
         }
 
-        public Task Handle(SagaUpdatedMessage message, IMessageHandlerContext context)
+        public async Task Handle(SagaUpdatedMessage message, IMessageHandlerContext context)
         {
-            var sagaSnapshot = SagaSnapshotFactory.Create(message);
+            await WarnOncePerEndpointPerDay(message);
 
-            return store.StoreSnapshot(sagaSnapshot);
+            if (auditQueueName is null || nextAuditQueueNameRefresh < DateTime.UtcNow)
+            {
+                await RefreshAuditQueue();
+            }
+
+            if (auditQueueName is null)
+            {
+                throw new Exception("Could not determine audit queue name to forward saga update message. The ServiceControl remote audit instance ");
```

Comment:
I think the message is incomplete?
```suggestion
                throw new Exception("Could not retrieve audit queue to forward saga update from audit connection info.");
```